core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1439,6 +1439,14 @@ private boolean needNewSubQuery(RelNode rel, Clause[] clauses) {
         return true;
       }
 
+      if (rel instanceof Project
```

Comment:
Do we also need to prevent merging projects in RelBuilder when converting `SqlNode` to `RelNode`?

```
LogicalProject(EXPR$0=[ROW_NUMBER() OVER (ORDER BY ROW_NUMBER() OVER (ORDER BY $1 ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)])
  JdbcTableScan(table=[[foodmart, product]])
```

Reply:
Yes, but that's covered in https://issues.apache.org/jira/browse/CALCITE-3079.